### ISSUE TITLE
Real browser reveraled over iframe,contentWindow method

### ISSUE THREAD

**Issue by crssi** [Snapper26/shapeshifter#5]
[https://browserleaks.com/javascript](https://browserleaks.com/javascript)

Pay attention to buttons `windows` and `iframe,contentWindow` and compare gathered `userAgent`, `oscpu` and `hardwareConcurrency`.

Another question: Is `buildID` aligned with spoofed version.

See topic [https://github.com/ghacksuserjs/ghacks-user.js/issues/203#issue-249490315](https://github.com/ghacksuserjs/ghacks-user.js/issues/203#issue-249490315)

**Comment by Snapper26** [comment]
Thank you. I am aware of this issue and plan to fix it in the next release.

**Comment by Thorin-Oakenpants** [comment]
> Is buildID aligned with spoofed version? I'm not spoofing it at this time so probably not

The buildID is revealed in the UA string and is different for every "build", so yes, it is aligned with every version, every dot version, for every OS

```js
/* 2697b: navigator.buildID (see gecko.buildID in about:config) reveals build time
 * down to the second which defeats user agent spoofing and can compromise OS etc
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=583181 ***/
```

**Comment by Snapper26** [comment]
Yep so I probably should just use 20100101 then? Also @crssi you mentioned about iframe.contentWindow still revealing the real browser attributes. Shape Shifter is a web extension which means that I can't use the obsolete/deprecated preferences service to set say general.useragent.override. This makes spoofing in iframe's quite tricky unless someone knows an easy way that I don't :)
